docs/context.md

```diff
@@ -1,22 +1,37 @@
 ---
-title: Context & Mixins
+title: Context
 ---
 
-## Context
-
 In order to use React's context, you need to create two things:
+
 1. The context itself
-2. A context provider component.
+2. A context provider component
 
 ```reason
-/** ContextProvider.re */
+/** as a separate file: ContextProvider.re */
+
+// 1. The context itself
 let themeContext = React.createContext("light");
 
+// 2. The provider
 include React.Context; // Adds the makeProps external
 let make = React.Context.provider(themeContext);
 ```
 
-That will give you a `ContextProvider` component you can use in your application later on. You'll do this like you'd normally would in any React application.
+```reason
+/** or inside a any other module */
+
+// 1. The context itself
+let themeContext = React.createContext("light");
+
+// 2. The provider component
+module ContextProvider = {
+  include React.Context; // Adds the makeProps external
+  let make = React.Context.provider(themeContext);
+};
+```
+
+That will give you a `ContextProvider` component you can use in your application later on, by wrapping any component with `ContextProvider`, to have access to the context value inside the component tree. To know more about Context, check the [official React documentation](https://legacy.reactjs.org/docs/context.html) and [when to use it](https://react.dev/learn/passing-data-deeply-with-context).
```

Comment:
should we update the legacy doc to https://react.dev/reference/react/createContext?